commons/service/meta.service.ts

```
@@ -89,33 +90,38 @@ export class MetaNetworkService implements IMetaNetworkService {
 			callable: Callable.CreateGroup,
 			inputParams: [adminId, groupId, groupInfo, country]
 		})
-			.signAndSend(
-				custodianKeyPair,
-				async ({ status, events = [] }: any) => {
-					if (status.isFinalized) {
-						const statusString = displayText(
-							status.asFinalized.toString()
-						);
-						enqueueSnackbar(`Transaction success ${statusString}`, {
-							variant: "success",
+			.signAndSend(custodianKeyPair, async ({ status }: any) => {
+				if (status.isFinalized) {
+					const blockHash = status.asFinalized.toString();
+					const { isSuccessful, message } = await txErrorHandler(
+						api,
+						blockHash
+					);
+
+					enqueueSnackbar(
+						isSuccessful ? "Transaction success" : message,
+						{
+							variant: isSuccessful ? "success" : "error",
```

in this case, we can just throw the error so that it can be caught in the `.catch()` block right?

---

Addressed on the other callable. Specifically for this ` Callable.CreateGroup` the implementation didn't use promise reject/resolve pattern, so instead of throwing error, displaying the message directly is more straight forward.